src/conjureFetch.ts

```diff
@@ -28,7 +29,24 @@ export async function conjureFetch<T>(
     }
   }
 
-  const response = await (fetchFn ?? fetch)(`${baseUrl}${servicePath}${url}`, {
+  const queryParams = Object.entries(params ?? {}).reduce(
```

Comment:
`conjureFetch` is now responsible for taking query params and constructing a suitable query for the URL. This omits null/undefined parameters as well as empty strings (this'll also omit empty arrays given `.join(",")`).